core/src/validator.rs

```diff
@@ -627,16 +629,19 @@ impl Validator {
             check_poh_speed(&genesis_config, None);
         }
 
-        if wait_for_supermajority(
+        let (failed, did_wait) = wait_for_supermajority(
```

Comment:
nit: `did_wait` -> `is_wait_unnecessary`, since in some cases like on hard forks, a wait didn't occur but is necessary (or even better `is_wait_necessary`, but then we have to flip the booleans around below)